Not working in RN v0.72.1

### What happened?

```
error Failed to install the app. Command failed with exit code 1: gradlew.bat tasks FAILURE: Build failed with an exception. * Where: Build file 'rnntest\node_modules\react-native-navigation\lib\android\app\build.gradle' line: 47 * What went wrong: Execution failed for task ':tasks'. > Could not create task ':react-native-navigation:testReactNative71DebugUnitTest'. > No signature of method: org.gradle.api.reporting.internal.TaskGeneratedSingleDirectoryReport.enabled() is applicable for argument types: (Boolean) values: [true] * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 
36s
> Task :gradle-plugin:pluginDescriptors > Task :gradle-plugin:processResources > Task :gradle-plugin:compileKotlin > Task :gradle-plugin:compileJava NO-SOURCE > Task :gradle-plugin:classes 
> Task :gradle-plugin:jar > Task :gradle-plugin:inspectClassesForKotlinIC > Task :tasks FAILED 6 actionable tasks: 6 executed.
```

### What was the expected behaviour?

Working in RN 0.72.1

### Was it tested on latest react-native-navigation?

- [X] I have tested this issue on the latest react-native-navigation release and it still reproduces.

### Help us reproduce this issue!

1. Create a latest (v0.72.1) blank RN project: `npx react-native@latest init rntest`
2. install RNN `yarn add react-native-navigation`
3. Run app `npm run android`

### In what environment did this happen?

React Native Navigation version: 7.33.3-hotfix.3
React Native version: 0.72.1
Has Fabric (React Native's new rendering system) enabled: yes

--- a/autolink/postlink/applicationLinker.js
+++ b/autolink/postlink/applicationLinker.js
@@ -98,6 +98,11 @@ class ApplicationLinker {
   }
 
   _extendNavigationHost(applicationContent) {
+    if (this._hasAlreadyLinkedNavigationHost(applicationContent)) {
+      warnn('   NavigationReactNativeHost is already used, skipping.');
+      return applicationContent;
+    }
+
     if (this._doesExtendReactNativeHost(applicationContent)) {
       debugn('   Changing host implementation to NavigationReactNativeHost');
       return applicationContent
@@ -106,11 +111,14 @@ class ApplicationLinker {
           'import com.facebook.react.ReactNativeHost;',
           'import com.facebook.react.ReactNativeHost;\nimport com.reactnativenavigation.react.NavigationReactNativeHost;'
         );
-    }
-
-    if (this._hasAlreadyLinkedNavigationHost(applicationContent)) {
-      warnn('   NavigationReactNativeHost is already used, skipping.');
-      return applicationContent;
+    } else if (this._doesExtendDefaultReactNativeHost(applicationContent)) {
+      debugn('   Changing host implementation to NavigationReactNativeHost');
+      return applicationContent
+        .replace('new DefaultReactNativeHost(this)', 'new NavigationReactNativeHost(this)')
+        .replace(
+          'import com.facebook.react.defaults.DefaultReactNativeHost;',
+          'import com.facebook.react.defaults.DefaultReactNativeHost;\nimport com.reactnativenavigation.react.NavigationReactNativeHost;'
+        );
     }
 
     throw new Error('There was a problem extending NavigationReactNativeHost().');
@@ -118,6 +126,10 @@ class ApplicationLinker {
 
   _doesExtendReactNativeHost(applicationContent) {
     return /\s*new ReactNativeHost\(this\)\s*/.test(applicationContent);
+  }
+
+  _doesExtendDefaultReactNativeHost(applicationContent) {
+    return /\s*new DefaultReactNativeHost\(this\)\s*/.test(applicationContent);
   }
 
   _hasAlreadyLinkedNavigationHost(applicationContent) {

--- a/autolink/postlink/gradleLinker.js
+++ b/autolink/postlink/gradleLinker.js
@@ -3,7 +3,7 @@ var path = require('./path');
 var fs = require('fs');
 var { warnn, errorn, logn, infon, debugn } = require('./log');
 var { insertString } = require('./stringUtils');
-var DEFAULT_KOTLIN_VERSION = '1.5.31';
+var DEFAULT_KOTLIN_VERSION = '1.7.10';
 // This should be the minSdkVersion required for RNN.
 var DEFAULT_MIN_SDK_VERSION = 21;
 
@@ -74,7 +74,7 @@ class GradleLinker {
       return contents;
     }
 
-    var match = /classpath\s*\(*["']com\.android\.tools\.build:gradle:/.exec(contents);
+    var match = /classpath\s*\(*["']com\.android\.tools\.build:gradle.*/.exec(contents);
     if (match) {
       debugn('   Adding Kotlin plugin');
       return insertString(

